### ISSUE TITLE
Issue creating a session with XCUITest 4.3.7

### ISSUE THREAD

**Issue by JuanCarlos-Arceredillo** [appium/appium#16784]
## The problem

Hi all,

I have tried updating the iOS driver with the command

appium driver update xcuitest

and after updating from the version 4.3.5 to the 4.3.7. It seems that at least this driver is correct.
Nevertheless, I'm trying just to open an App on an iOS simulator with the Appium Inspector and I'm facing several issues.

The simulator isn't opened as with Appium server GUI.
Appium log shows this error message Encountered internal error running command: Error: The simulator ********-****-****-****-************ has failed to finish booting after 120s.
The appium Inspector reports a timeout error.
After a while, it seems that Appium tries again to install the WDA on the simulator and it starts the common connecting retries like:

Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body connect ECONNREFUSED 127.0.0.1:8100

But instead to opening the simulator or connecting as it's the normal workflow with Appium 1.X.
Maybe it's all related to the issue that the simulator is not opened, but I'm not sure. I have also tried opening manually the simulator but the behavior is the same.

## Environment

* Appium version (or git revision) that exhibits the issue: v2.0.0-beta.30
* Last Appium version that did not exhibit the issue (if applicable): I've tried Appium Server GUI v1.22 without any issue.
* Desktop OS/version used to run Appium: MacOS
* Mobile platform/version under test: iOS 
* Real device or emulator/simulator: Simulator (iPhone 11 Pro Max with iOS 14.2)

## Code To Reproduce Issue [ Good To Have ]

I've used the following capabilities for the test.

`
{
  "platformName": "iOS",
  "deviceName": "iPhone 11 Pro Max",
  "automationName": "XCUITest",
  "language": "es",
  "locale": "ES",
  "platformVersion": "14.2"
}
`

Thank you for your support.

**Comment by mykola-mokhnach** [comment]
Please provide the full server log. We cannot help much without it.

> Maybe it's all related to the issue that the simulator is not opened, but I'm not sure. I have also tried opening manually the simulator but the behavior is the same.

If a simulator times out after manual launch then it is most likely not an Appium issue. Try to reset/recreate it or bump the launch timeout in capabilities (or both).

**Comment by JuanCarlos-Arceredillo** [comment]
Hi @mykola-mokhnach ,

Here you can find the complete server log. I have seen that the log detects iOS 15.2 (the newest that I have installed) even when the capabilities have a different value. Nevertheless, have tried again with the 15.2 and the behavior is the same.

It seems that there is an issue installing the WDA, but I'm not sure.

[errorlog.txt](https://github.com/appium/appium/files/8504210/errorlog.txt)

**Comment by mykola-mokhnach** [comment]
> [Xcode] 2022-04-18 09:08:19.779610+0200 WebDriverAgentRunner-Runner[25754:134138] Failed to initialize for UI testing: Error Domain=com.apple.dt.XCTest.XCTFuture Code=1000 "Timed out while loading Accessibility." UserInfo={NSLocalizedDescription=Timed out while loading Accessibility.}
> [Xcode] 
> [Xcode] 
> [Xcode] 
> [Xcode] *** If you believe this error represents a bug, please attach the result bundle at /Users/TestUser/Library/Developer/Xcode/DerivedData/WebDriverAgent-bjoyioxgytkismchxtsyrzasdjzn/Logs/Test/Test-WebDriverAgentRunner-2022.04.18_09-06-25-+0200.xcresult
> [Xcode] 
> [Xcode] 
> [debug] [XCUITestDriver@0223 (13ae1b0d)] Matched '/status' to command name 'getStatus'
> [debug] [XCUITestDriver@0223 (13ae1b0d)] Proxying [GET /status] to [GET http://127.0.0.1:8100/status] with no body
> [XCUITestDriver@0223 (13ae1b0d)] connect ECONNREFUSED 127.0.0.1:8100
> [Xcode] 2022-04-18 09:08:20.048 xcodebuild[25163:135700]  IDETestOperationsObserverDebug: Failure collecting logarchive: Error Domain=NSCocoaErrorDomain Code=4099 "The connection to service on pid 0 named com.apple.testmanagerd.control was invalidated." UserInfo={NSDebugDescription=The connection to service on pid 0 named com.apple.testmanagerd.control was invalidated.}
> [Xcode] 

This seems to be some Xcode error, I don't think it has to do with xcuitest driver. Have you tried to run it on another machine?

**Comment by JuanCarlos-Arceredillo** [comment]
Sorry, I don't have more machines to test at the moment.

The weird thing is that if I create a session from the Inspector on the Appium Server GUI 1.22, it works fine, it installs the AWD app correctly and creates the session.

Moreover, in the last try I have seen that if I install AWD with Appium Server GUI 1.22 and launch a second test on Appium 2.0 immediately, the inspector connects with the simulator. Maybe there is an issue with the AWD installation process.

**Comment by KazuCocoa** [comment]
> [debug] [XCUITestDriver@0223 (13ae1b0d)] Beginning test with command 'xcodebuild build-for-testing test-without-building -project /Users/TestUser/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent/WebDriverAgent.xcodeproj -scheme WebDriverAgentRunner -destination id=2F938B7D-103D-40B1-B9DB-6B7465FA0815 IPHONEOS_DEPLOYMENT_TARGET=14.2 GCC_TREAT_WARNINGS_AS_ERRORS=0 COMPILER_INDEX_STORE_ENABLE=NO' in directory '/Users/TestUser/.appium/node_modules/appium-xcuitest-driver/node_modules/appium-webdriveragent'

The xcodebuild is how WDA is being installed. What happens when you run it by manual?

The error message seems https://developer.apple.com/forums/thread/653570

**Comment by JuanCarlos-Arceredillo** [comment]
Thank you @KazuCocoa for your comments, I have tried to run the xcodebuild manually and I've found the issue. My current user doesn't have an administrative role, so to install appium 2.0 I had to change to a sudoer user.

At this point everything is clear, but if I install the drivers and plugins with this admin user, they'll be installed in the .appium folder inside the current user folder:

`/Users/AdminUser/.appium/node_modules/.......`

and if I try to run appium with my normal user, these folders don't exist.

`/Users/TestUser/.appium/node_modules/.......`

I'm not sure if it's an issue or if the documentation must advice about this possible situation.

Anyway... after installing all plugins & drivers with my common user... everything is working as expected. 
Thank you for your support
